### PR TITLE
Improve server-side of world_sync

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

File: randovania/server/multiplayer/world_api.py [hunk 1]
```diff
@@ -174,42 +209,30 @@ def world_sync(sio: ServerApp, request: ServerSyncRequest) -> ServerSyncResponse
...
-            failed_syncs[uid] = error.ServerError()
+                response, session_id, new_worlds_to_update = sync_one_world(sio, user, uid, world_request)
+
+                if response is not None:
+                    world_details[uid] = response
+
+                if session_id is not None:
+                    sessions_to_update_meta.add(session_id)
+
+                worlds_to_update.update(new_worlds_to_update)
```
Comment (ThanatosGit) [inline randovania/server/multiplayer/world_api.py:225]: Why we need to do this again? This is already done in `sync_one_world` in `if world_request.collected_locations`.

File: randovania/server/multiplayer/world_api.py [hunk 2]
```diff
@@ -174,42 +209,30 @@ def world_sync(sio: ServerApp, request: ServerSyncRequest) -> ServerSyncResponse
     sessions_to_update_meta = set()
 
     for uid, world_request in request.worlds.items():
-        try:
+        with sentry_sdk.start_transaction(op="message", name="sync_one_world") as span:
+            span.set_tag("world_id", uid)
+            span.set_data("message.error", 0)
```
Comment (ThanatosGit) [inline randovania/server/multiplayer/world_api.py:214]: L235 sets it anyways to `span.set_data("message.error", failed_syncs.get(uid, 0))`. It's redudant?

**Review by ThanatosGit — COMMENTED** [review]
Just getting slowly familiar with parts I don't know. Just some comments with high potential that I didn't understood it :p

File: randovania/server/multiplayer/world_api.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/server/multiplayer/world_api.py:214]: Paranoia with making sure it's set even if some random exception happens.

File: randovania/server/multiplayer/world_api.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/server/multiplayer/world_api.py:225]: Two different `worlds_to_update` variables!

**Pull request merged**